torch/csrc/jit/ir.h

```diff
@@ -69,6 +69,65 @@ struct SourceLocation {
   std::string python_traceback;
 };
 
+// Scope is a node of a trie that represents the tree of nested scopes.
+// Individual scopes are pushed and popped from TracingState, which holds a
+// pointer to the current scope. Each Node in Graph holds a pointer
+// to the scope that was current when the node was created.
+// The trie never needs to shrink, it only grows until it is disposed
+// of when TracingState is deallocated. Hence, pointers to scopes held by nodes
+// will always be valid as long as TracingState is alive.
+struct Scope {
+private:
+  Scope* parent_;
+  Symbol name_;
+  std::unordered_set<std::unique_ptr<Scope> > children_;
```

Comment:
The only reason is that you could theoretically `push` a scope that is already there, with the effect that you make a previous scope become current again. With a vector, we simply have to check before we insert to avoid duplication - this is done infrequently and on a small vector, so it makes perfect sense.

Reply:
Ohh, right. let's keep it as a set then!